src/shared/util.js

```diff
@@ -591,13 +591,9 @@ function string32(value) {
                              (value >> 8) & 0xff, value & 0xff);
 }
 
+// NOTE: This name is misleading as it rounds the result up
 function log2(x) {
-  var n = 1, i = 0;
-  while (x > n) {
-    n <<= 1;
-    i++;
-  }
-  return i;
+  return Math.ceil(Math.log(x) * Math.LOG2E);
```

Comment:
I tracked down all the paths log2() output might go and it seems like the only place when I'm not certain that input > 0 is where the result, saved as symbolCodeLength, is ultimately passed to decodeIAID as a third argument. There are only two places like this, so we could, just to be sure, change

`var symbolCodeLength = log2(inputSymbols.length)`

to

`var symbolCodeLength = inputSymbols.length && log2(inputSymbols.length)`